pyiron_base/jobs/flex/executablecontainer.py

```diff
@@ -115,8 +118,26 @@ def get_calculate_function(self) -> callable:
         Returns:
             callable: calculate() functione
         """
+
+        def get_combined_write_input_funct(input_job_dict, write_input_funct=None):
+            def write_input_combo_funct(working_directory, input_dict):
+                write_input_files_from_input_dict(
+                    input_dict=input_job_dict,
+                    working_directory=working_directory,
+                )
```

Comment:
It might be easier to follow if we add this call in [CalculateFunctionCaller](https://github.com/pyiron/pyiron_base/blob/51ba6c28a4dddff17bd75ca29971d8659bc652ba/pyiron_base/jobs/job/runfunction.py#L134) directly or would this break something else?

Reply:
I looked at it. In the end it comes down to the question: Do developers of `GenericJob` classes remember to add the `super().get_input_parameter_dict()` call to their `get_input_parameter_dict()` function? We could move around this by implementing an `_get_input_parameter_dict()` function but I am not sure if that is more intuitive for scientists. At the same time I like that all input files are now handled via the same interface. 

For the specific case of the `wrap_executable()` function it is harder to add the `super()` call so here we have the internal workaround in the internal class which wraps the executable. 

In summary, I am reluctant to implement this functionality directly in the `CalculateFunctionCaller` as it would break the current scheme in which the content of all files is communicated via the `get_input_parameter_dict()` function.